src/test/java/org/minions/devfund/bruno/html/HtmlValidatorTest.java

```diff
@@ -0,0 +1,196 @@
+
+import static org.junit.Assert.assertTrue;
+import static org.minions.devfund.bruno.html.HtmlValidator.isValidHtml;
+
+/**
+ * Tests for html validator.
+ */
+public class HtmlValidatorTest {
+
+    /**
+     * Test file 1.
+     */
+    @Test
+    public void testFile1() {
+        try {
+            Stack<HtmlTag> var1 = this.validateTestFile("test1.html");
+            if (var1 == null) {
+                Assert.fail("isValidHtml returns null for a valid HTML file input.");
+            }
+            assertTrue("isValidHtml should return an empty stack for a valid HTML file input.",
+                    var1.isEmpty());
+        } catch (Exception var2) {
+            Assert.fail("isValidHtml throws " + var2 + " for valid HTML file input");
+        }
+    }
+
+    /**
+     * Test file 2.
+     */
+    @Test
+    public void testFile2() {
+        try {
+            Stack<HtmlTag> var1 = this.validateTestFile("test2.html");
+            if (var1 == null) {
+                Assert.fail("isValidHtml returns null when file ends without closing tags");
+            }
+            Stack<HtmlTag> var2 = new Stack<>();
+            var2.push(new HtmlTag("html", true));
+            var2.push(new HtmlTag("b", true));
+            Assert.assertEquals(
+                    "Should return Stack with unmatched elements when file ends without closing tags",
+                    var2, var1);
+        } catch (Exception var3) {
+            Assert.fail("isValidHtml throws " + var3 + " when file ends without closing tags");
+        }
+    }
+
+    /**
+     * Test file 3.
+     */
+    @Test
+    public void testFile3() {
+        try {
+            Stack<HtmlTag> var1 = this.validateTestFile("test3.html");
+            if (var1 == null) {
+                Assert.fail("isValidHtml returns null when tags are closed in incorrect order");
+            }
+            Stack<HtmlTag> var2 = new Stack<>();
+            var2.push(new HtmlTag("b", true));
+            var2.push(new HtmlTag("i", true));
+            Assert.assertEquals(
+                    "Should return Stack with unmatched elements when tags are closed in incorrect order",
+                    var2, var1);
+        } catch (Exception var3) {
+            Assert.fail("isValidHtml throws " + var3 + " when tags are closed in incorrect order");
+        }
+
+    }
+
+    /**
+     * Test file 4.
+     */
+    @Test
+    public void testFile4() {
+        try {
+            Stack<HtmlTag> var1 = this.validateTestFile("test4.html");
+            Assert.assertNull("Should return null when encountering closing tag without matching opening tag", var1);
+        } catch (Exception var2) {
+            Assert.fail("isValidHtml throws " + var2 + " when encountering closing tag without matching opening tag");
+        }
+    }
+
+    /**
+     * Test file 5.
+     */
+    @Test
+    public void testFile5() {
+        try {
+            Stack<HtmlTag> var1 = this.validateTestFile("test5.html");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "Stack" by an unsynchronized one such as "Deque". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)